provers/stark/src/examples/read_only_memory.rs

```diff
@@ -0,0 +1,435 @@
+use std::marker::PhantomData;
+
+use crate::{
+    constraints::{
+        boundary::{BoundaryConstraint, BoundaryConstraints},
+        transition::TransitionConstraint,
+    },
+    context::AirContext,
+    frame::Frame,
+    proof::options::ProofOptions,
+    trace::TraceTable,
+    traits::AIR,
+};
+use lambdaworks_crypto::fiat_shamir::is_transcript::IsTranscript;
+use lambdaworks_math::field::traits::IsPrimeField;
+use lambdaworks_math::{
+    field::{element::FieldElement, traits::IsFFTField},
+    traits::ByteConversion,
+};
+
+/// This condition ensures the continuity in a read-only memory structure, preserving strict ordering.
+/// Equation based on Cairo Whitepaper section 9.7.2
+#[derive(Clone)]
+struct ContinuityConstraint<F: IsFFTField> {
+    phantom: PhantomData<F>,
+}
+
+impl<F: IsFFTField> ContinuityConstraint<F> {
+    pub fn new() -> Self {
+        Self {
+            phantom: PhantomData,
+        }
+    }
+}
+
+impl<F> TransitionConstraint<F, F> for ContinuityConstraint<F>
+where
+    F: IsFFTField + Send + Sync,
+{
+    fn degree(&self) -> usize {
+        2
+    }
+
+    fn constraint_idx(&self) -> usize {
+        0
+    }
+
+    fn end_exemptions(&self) -> usize {
+        // NOTE: We are assuming that the trace has as length a power of 2.
+        1
+    }
+
+    fn evaluate(
+        &self,
+        frame: &Frame<F, F>,
+        transition_evaluations: &mut [FieldElement<F>],
+        _periodic_values: &[FieldElement<F>],
+        _rap_challenges: &[FieldElement<F>],
+    ) {
+        let first_step = frame.get_evaluation_step(0);
+        let second_step = frame.get_evaluation_step(1);
+
+        let a_sorted_0 = first_step.get_main_evaluation_element(0, 2);
+        let a_sorted_1 = second_step.get_main_evaluation_element(0, 2);
+        // (a'_{i+1} - a'_i)(a'_{i+1} - a'_i - 1) = 0 where a' is the sorted address
+        let res = (a_sorted_1 - a_sorted_0) * (a_sorted_1 - a_sorted_0 - FieldElement::<F>::one());
+
+        // The eval always exists, except if the constraint idx were incorrectly defined.
+        if let Some(eval) = transition_evaluations.get_mut(self.constraint_idx()) {
+            *eval = res;
+        }
+    }
+}
+
+#[derive(Clone)]
```

Comment:
this annotation should be placed after the documentation

Reply:
ok, [done](https://github.com/lambdaclass/lambdaworks/pull/940/commits/59c8b2681bf85c79d3bd5c484569d4ceb3a63612).